knimin/lib/data_access.py

```diff
@@ -2425,3 +2553,38 @@ def _revert_ready(self, barcodes):
                  SET results_ready = NULL
                  WHERE barcode IN %s"""
         self._con.execute(sql, [tuple(barcodes)])
+
+    def _add_new_samples(self, sample_ids):
+        """Test helper to create new samples to test with
+
+        Parameters
+        ----------
+        sample_ids : list of str
+            Sample ids to create
+        """
+        sql = """INSERT INTO pm.sample (sample_id) VALUES (%s)"""
+        self._con.executemany(sql, [[x] for x in sample_ids])
+
+    def _delete_new_samples(self, sample_ids):
+        """Test helper to delete the samples that were just created
+
+        Parameters
+        ----------
+        sample_ids : list of str
+            Sample ids to delete
+        """
+        sql = """DELETE FROM pm.sample WHERE sample_id = %s"""
```

Comment:
@qiyunzhu Note that this function will fail if the sample is attached to a plate - thus we need to perform some validation before "blindly" remove the sample. We probably don't want to let the user remove the sample if it is attached to a plate.